[tensorexpr] Add PYTORCH_TENSOREXPR_DONT_FUSE env variable to disable fusion on specified operators - fixed #50757

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55650 [tensorexpr] Add PYTORCH_TENSOREXPR_DONT_FUSE env variable to disable fusion on specified operators - fixed #50757**

Differential Revision: [D27667232](https://our.internmc.facebook.com/intern/diff/D27667232)

huiguoo committed:
[tensorexpr] Add PYTORCH_TENSOREXPR_DONT_FUSE env variable to disable fusion on specified operators - fixed #50757

[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -712,6 +713,7 @@ class TensorExprFuser {
   // Merge fusible nodes into subgraphs in prim::TensorExprGroup nodes.
   void createFusionGroups(Block* block) {
     bool any_changed = true;
+    parseTENotFuseOption();
```
This function initializes the `operators_not_to_fuse` member variable, right?  I'd recommend calling it in the constructor rather than here in one of the member functions.

bertmaher commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -1159,9 +1163,33 @@ class TensorExprFuser {
+  // operators that are separated by ':'. e.g.,
+  // 'PYTORCH_TENSOREXPR_DONT_FUSE="clamp:mul:add"' disables fusion on
+  // aten::clamp, aten::mul and aten::add.
+  std::set<NodeKind> parseTENotFuseOption() {
```
Since this is initializing a member, I'd probably make it `void` and not return anything.  I guess alternatively you could have it construct a set as a local variable, return that, and put it in the initialization list of the constructor.  I'm just wary of a function returning a value that is probably thrown away :-)

bertmaher reviewed (approved):
Seems like a useful feature!  I've just a couple minor comments.

bertmaher reviewed (commented):
Oh, one more thing -- could you please write a more detailed Summary and Test Plan in the commit message (I guess for this an acceptable test plan is probably "tried it" but whatever :-p )

huiguoo committed:
Update on "[tensorexpr] Add PYTORCH_TENSOREXPR_DONT_FUSE env variable to disable fusion on specified operators - fixed #50757"


Differential Revision: [D27667232](https://our.internmc.facebook.com/intern/diff/D27667232)

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Add PYTORCH_TENSOREXPR_DONT_FUSE env variable to disable fusion on specified operators - fixed #50757"


Differential Revision: [D27667232](https://our.internmc.facebook.com/intern/diff/D27667232)

[ghstack-poisoned]
